src/FFSetup.cpp

```diff
@@ -380,14 +380,26 @@ void Dihedral::Read(Reader &param, std::string const &firstVar) {
   if (index == 0) {
     // set phase shift for n=0 to 90 degree
     // We will have C0 = Kchi (1 + cos(0 * phi + 90)) = Kchi
-    //this avoids double counting the C0 (constant offset) term, which is used force fields like TraPPE
+    // this avoids double counting the C0 (constant offset) term, which is used
+    // in force fields like TraPPE
     def = 90.00;
   }
   Add(merged, coeff, index, def);
   last = merged;
 }
 void Dihedral::Add(std::string const &merged, const double coeff,
                    const uint index, const double def) {
+  // Check for (and skip) duplicate periodicities for the same dihedral
+  bool duplicate = false;
+  for (auto it = n[merged].begin(); it != n[merged].end(); ++it) {
+    duplicate |= *it == index;
+  }
+
+  if (duplicate) {
+    std::cout << "Warning: Skipping duplicate periodicity of " << index
```

Comment:
I tested is and it Looks like it is working as intended.

Reply:
I updated the patch so that it throws an error if the parameters differ by more than 0.001. I have to imagine that would be a mistake. Since it only proceeds if they match, then which one we save doesn't matter.

So, we do the same thing as NAMD if the parameters match except we also issue a warning, and we terminate with an error if the parameters differ.